src/dsp_tools/commands/excel2json/new_lists/make_new_lists.py

```diff
@@ -59,55 +60,50 @@ def new_excel2lists(
     return finished_lists, True
 
 
-def _parse_files(excelfolder: Path | str) -> list[ExcelSheet]:
+def _parse_files(excelfolder: Path | str) -> dict[str, dict[str, pd.DataFrame]]:
     file_names = [file for file in Path(excelfolder).glob("*list*.xlsx", case_sensitive=False) if _non_hidden(file)]
-    all_sheets = []
+    df_dict = {}
     for file in file_names:
-        all_sheets.extend(
-            [
-                ExcelSheet(excel_name=str(file), sheet_name=name, df=df)
-                for name, df in read_and_clean_all_sheets(file).items()
-            ]
-        )
-    return all_sheets
+        df_dict[str(file)] = read_and_clean_all_sheets(file)
+    return df_dict
 
 
-def _non_hidden(path: Path) -> bool:
-    return not regex.search(r"^(\.|~\$).+", path.name)
+def _prepare_sheets(df_dict: dict[str, dict[str, pd.DataFrame]]) -> list[ExcelSheet]:
+    all_sheets: list[ExcelSheet] = []
+    for file, sheets in df_dict.items():
+        all_sheets.extend(_prepare_one_sheet(df, file, sheet_name) for sheet_name, df in sheets.items())
+    make_all_excel_compliance_checks(all_sheets)
+    return _construct_ids(all_sheets)
 
 
-def _prepare_dfs(sheet_list: list[ExcelSheet]) -> list[ExcelSheet]:
-    sheet_list = _add_id_optional_column_if_not_exists(sheet_list)
-    make_all_excel_compliance_checks(sheet_list)
-    return _construct_ids(sheet_list)
+def _prepare_one_sheet(df: pd.DataFrame, filename: str, sheet_name: str) -> ExcelSheet:
+    columns = get_column_info(df.columns)
+    df = add_optional_columns(df, {"id (optional)"})
+    return ExcelSheet(excel_name=filename, sheet_name=sheet_name, col_info=columns, df=df)
 
 
-def _add_id_optional_column_if_not_exists(sheet_list: list[ExcelSheet]) -> list[ExcelSheet]:
```

Comment:
I have a generic function that does the same. This occurs when the sheets are prepared.